assignment/scripts/function-practice.js

```diff
@@ -49,34 +49,58 @@ console.log( 'isPositive - should say false', isPositive(-3) );
 
 // 6. Function to return the _last_ item in an array. If the 
 //    array is empty, return `undefined`.
-function getLast( array ) {
-
+function getLast(array) {
+  if (array.length > 0) {
+    return array.length - 1;
+  } 
+  return 'undefined';
 }
+console.log(getLast([1, 2, 5]));
+console.log(getLast([]));
+
 
 // 7. Function to find a value in an array. Return true if the 
 //    value is found and false otherwise. Use a loop;
 //    DO NOT use Array.includes, Array.indexOf, or Array.find 
-function find( value, array ){
-  
-}
 
+function find(value, array) {
+  for (let item of array) {
+    if (item === value) {
+      return true;
+    }
+  }
+  return false;
+}
+console.log(find(4, [1, 2, 4]));
+console.log(find(4, [1, 2]));
 // ----------------------
 // Stretch Goals
 // ----------------------
 // 8. Function to check if a letter is the first letter in a 
 //    string. Return true if it is, and false otherwise
 function isFirstLetter(letter, string) {
-
+  let item = letter;
+  if (item[0] === string[0]) {
+      return true;
+    }
+  return false;
 }
-console.log( 'isFirstLetter - should say true', isFirstLetter('a', 'apple') );
-console.log( 'isFirstLetter - should say false', isFirstLetter('z', 'apple') );
+console.log('isFirstLetter - should say true', isFirstLetter('a', 'apple'));
+console.log('isFirstLetter - should say false', isFirstLetter('z', 'apple'));
 
 // 9. Function to return the sum of all numbers in an array
-function sumAll( ) {
-  let sum = 0
+
+// Almost had it finished
+function sumAll(array) {
+  let sum = 0;
   // TODO: loop to add items
+  for (let i = 0; i <= array.length; i++) {
+  sum = i++;
```

Comment:
There are a couple of issue with this implementation. It will produce the same number for an array of a given length, regardless of what numbers it contains.

```js
sumAll([3]);  // 2
sumAll([7]);  // 2

sumAll([3, 5, 7]);  // 6
sumAll([9, 9, 9]);  // 6
```

Did you notice this in your testing? Any idea what the cause is?